app/src/main/java/com/android/feedme/model/viewmodel/GenerateViewModel.kt

```diff
@@ -0,0 +1,88 @@
+package com.android.feedme.model.viewmodel
+
+import androidx.lifecycle.ViewModel
+import com.android.feedme.model.data.Profile
+import com.android.feedme.model.data.Recipe
+import com.android.feedme.model.data.RecipeRepository
+import com.google.firebase.firestore.DocumentSnapshot
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+/**
+ * A class that generates the view model for the generate screen
+ *
+ * This class provides the link between the generate screen and the rest of the code. It can be used
+ * in order to generate the recipes
+ */
+class GenerateViewModel : ViewModel() {
+  private val recipeRepository = RecipeRepository.instance
+
+  private val _generatedRecipes = MutableStateFlow<List<Recipe>>(emptyList())
+  val generatedRecipes = _generatedRecipes.asStateFlow()
+
+  private var lastRecipe: DocumentSnapshot? = null
+  private var _ingredientsIds: List<String> = emptyList()
+  private var _profile: Profile = Profile()
+
+  private val _isStrict = MutableStateFlow(false)
+
+  /**
+   * A function that fetches the generated recipes
+   *
+   * @param ingredientIds: the unique IDs of the recipes we want to fetch
+   * @param profile: the profile of the user
+   */
+  fun fetchGeneratedRecipes(ingredientIds: List<String>, profile: Profile) {
+    _ingredientsIds = ingredientIds
+    _profile = profile
+
+    if (_isStrict.value) {
```

Comment:
The logic is reversed, by default, the toggle button is on "strict" and it's seem the value is true as when sliding to "extra" is does the strict search.

Reply:
Ohhh my bad, nice catch !